### ISSUE TITLE
Subdocuments should not have parent document's instance methods

### ISSUE THREAD

**Issue by wlingke** [Automattic/mongoose#2149]
If you define a Schema like so:

```
{
    field: {
        stuff: String
    }
}
```

and an instance method:

```
Schema.methods.test = function(){
    console.log('instance method')
}
```

Then create a new document and get the 'field' object as so:

```
var m = new Model()
var field = m.field;
```

`field` can access the Schema's instance variables:

```
field.test() /// 'instance method'
```

I dont know if this was intentional but it doesnt seem to make sense to me that it does

**Comment by ti0ma** [comment]
+1 here,

I'm using toObject method every time I need to iterate over keys or check if an object/array is empty.

Some news about that?

**Comment by YagoQuinoy** [comment]
+1

**Comment by vkarpov15** [comment]
This was fixed in [4.12.0](https://github.com/Automattic/mongoose/blob/master/History.md#4120--2017-10-02) with #5586, etc.
